diff --git a/vllm_metax/v1/attention/backends/mla/flashmla_sparse.py b/vllm_metax/v1/attention/backends/mla/flashmla_sparse.py
--- a/vllm_metax/v1/attention/backends/mla/flashmla_sparse.py
+++ b/vllm_metax/v1/attention/backends/mla/flashmla_sparse.py
@@ -153,6 +153,7 @@
     num_decodes: int = 0
     num_prefills: int = 0
     num_decode_tokens: int = 0
+    decode_max_query_len: int = 0
     seq_lens: torch.Tensor | None = None
     prefill_max_seq_len: int = 0
     prefill: MLACommonPrefillMetadata | None = None
